messagingpush/src/main/java/io/customer/messagingpush/activity/NotificationClickReceiverActivity.kt

```diff
@@ -43,98 +34,23 @@ class NotificationClickReceiverActivity : Activity(), TrackableScreen {
     }
 
     private fun handleIntent(data: Intent?) {
-        kotlin.runCatching {
-            val payload: CustomerIOParsedPushPayload? =
-                data?.extras?.parcelable(NOTIFICATION_PAYLOAD_EXTRA)
-            if (payload == null) {
-                logger.error("Payload is null, cannot handle notification intent")
+        if (data == null) {
+            // This should never happen ideally
+            logger.error("Intent is null, cannot process notification click")
+        } else {
```

Comment:
```suggestion
        if (data == null) {
            // This should never happen ideally
            logger.error("Intent is null, cannot process notification click")
            return 
        }      
```

I think the guard design pattern would be nice here instead of using `if/else`. Not a blocker, just nit-picky.

Reply:
We unfortunately can't use guard here because it should finish activity in all cases before exiting from this method.